src/Compilers/Core/Portable/CommandLine/CommonCompiler.CompilerEmitStreamProvider.cs

```diff
@@ -20,16 +20,28 @@ private sealed class CompilerEmitStreamProvider : Compilation.EmitStreamProvider
             private readonly CommonCompiler _compiler;
             private readonly string _filePath;
             private Stream _streamToDispose;
+            private readonly DiagnosticBag _diagnostics;
 
-            internal CompilerEmitStreamProvider(CommonCompiler compiler, string filePath)
+            internal CompilerEmitStreamProvider(
+                CommonCompiler compiler,
+                string filePath,
+                DiagnosticBag diagnostics)
             {
                 _compiler = compiler;
                 _filePath = filePath;
+                _diagnostics = diagnostics;
             }
 
             public void Dispose()
             {
-                _streamToDispose?.Dispose();
+                try
+                {
+                    _streamToDispose?.Dispose();
+                }
+                catch (Exception e)
+                {
+                    ReportOpenFileDiagnostic(_diagnostics, e);
```

Comment:
>_diagnostics [](start = 45, length = 12)

It is hard to tell if anyone is going to check this bag after this method is called. I assume many calls are implicit. I think we should consider to not use IDisposable pattern at all, Implement Close method and call it explicitly.